low-level/files/opt/seagate/sspl/setup/sspl_setup.py

```diff
@@ -182,20 +183,47 @@ def process(self):
 
 
 class TestCmd(Cmd):
-    """Starts test based on plan (sanity | alerts)."""
+    """Starts test based on plan:
+    (sanity|alerts|self_primary|self_secondary)."""
 
     name = "test"
+    test_plan_found=False
+    sspl_test_plans = ["sanity", "alerts", "self_primary","self_secondary"] 
 
     def __init__(self, args):
         super().__init__(args)
 
     def validate(self):
-        # Common validator classes to check Cortx/system wide validator
-        pass
+        if self.args:
+            i=0
+            while i < len(self.args):
+                # Provision for --config <global_config_url> for future use-case.
+                if self.args[i] == "--config":
+                    try:
+                        global_config = self.args[i+1]
+                        Conf.load('global_config', global_config)
+                    except (IndexError):
+                        raise SetupError(1, "global_config_url not specified. Please check usage")
```

Comment:
'pass' here, dont error out on missing config, just print and proceed as we are not really using it today. I mentioned that in previous comment

Reply:
@cdeshmukh the above code will raise error only if --config flag is parsed and the url i.e(next arg[i+1]) is empty.
Currently we will not pass the config flag, so it will never raise the exception..
Do we still need to remove this block of code?

Reply:
agree, we can keep the code then.